sympy/physics/mechanics/functions.py

```diff
@@ -422,8 +444,17 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
-    return set([i for i in expression.atoms(AppliedUndef, Derivative) if
-            i.free_symbols == t_set]) - exclude_set
+    try:
+        if isinstance(expression, Vector):
+            expr=expression.to_matrix(reference_frame)
+            return set([i for i in expr.atoms(AppliedUndef, Derivative) if
+                    i.free_symbols == t_set]) - exclude_set
+    except TypeError:
+        print(" function missing one required argument of type ReferenceFrame ")
+        return
+    if reference_frame is None :
+        return set([i for i in expression.atoms(AppliedUndef, Derivative) if
+                i.free_symbols == t_set]) - exclude_set
```

Comment:
This line is repeated and should be pulled out of the type checking code.

Reply:
I have changed it to:

if isinstance(expression, Vector) and reference_frame is None :
            raise TypeError("You must provide a reference frame when passing in a vector expression.")
            return
    if isinstance(expression, Vector) :
        expression=expression.to_matrix(reference_frame)
    return set([i for i in expression.atoms(AppliedUndef, Derivative) if
            i.free_symbols == t_set]) - exclude_set

without using any try exception blocks, does it seem fine?

Reply:
I like this better. Write unit tests to test whether it is fine. You don't need the first return statement.